src/virtual-machine.js

```diff
@@ -577,6 +578,15 @@ class VirtualMachine extends EventEmitter {
         // Filter events by type, since monitor blocks only need to listen to these events.
         // Monitor blocks shouldn't be destroyed when flyout blocks are deleted.
         if (['create', 'change'].indexOf(e.type) !== -1) {
+            let blockType = e.blockId.split('_');
+            blockType = blockType[blockType.length - 1];
```

Comment:
Can you use the opcode that is attached to the block instead of parsing out the ID? Block containers (like `runtime.monitorBlocks`) have `getBlock(id)` and `getOpcode(block)` functions that could help with this?

Reply:
You're right, the data was there all along.